packages/app/components/Post/Post.tsx

```diff
@@ -132,12 +132,17 @@ const Post = ({
   const upvoteHandler = useCallback(
     (_id: string) => {
       if (isAuthenticated) {
-        return mutate({ postId: _id });
+        if(isPassportOwner){
+          return mutate({ postId: _id });
```

Comment:
This check should probably be at the api upvote hander

Reply:
I was thinking that by storing the ownership status in global state after it's been derived either by the verify or passport routes, we minimize the number of external requests we need to make for checking token ownership status.  What do you think?

Reply:
that's a good idea for the front-end side, but we still should authorize server-side